sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md

```diff
@@ -678,143 +694,245 @@ class ServiceOperations:
+from azure.core.polling import PollingMethod
+from azure.core.polling.base_polling import BadResponse
+from azure.core.exceptions import ResourceNotFoundError
+JSON = MutableMapping[str, Any]
+
+
 class CustomLROPolling(PollingMethod):
-    def __init__(self, polling_interval=30, **kwargs) -> None:
+    def __init__(self, polling_interval: float = 30, **kwargs: Any) -> None:
+        """Creates a custom polling method which polls until a file is uploaded.
+        The operation is considered to have reached a terminal state once a successful GET
+        is done on the file (e.g. no ResourceNotFoundError is raised).
+
+        :param polling_interval: The amount of time to wait between polls. This fictitious service does not
+            use retry-after so we will default to this value.
+        :param kwargs: Any operation-specific keyword arguments that should be passed into the GET call.
+        """
         self._polling_interval = polling_interval
-        self._resource = None
-        self._finished = False
-        self._status = "InProgress"
         self._kwargs = kwargs
 
-    def initialize(self, client, initial_response, deserialization_callback):
-        """Set the initial status of this LRO.
-        """ 
+    def initialize(self, client: Any, initial_response: PipelineResponse, deserialization_callback: Callable) -> None:
+        """Set the initial status of this LRO, verify that we can poll, and 
+        initialize anything necessary for polling.
+
+        :param client: An instance of a client. In this example, the generated client.
+        :param initial_response: In this example, the PipelineResponse returned from the initial call.
+        :param deserialization_callback: A callable to transform the final result before returning to the caller.
+        """
+        
+        # verify we have the information to poll
+        response = initial_response.http_response.json()
+        if response.get("file_id", None) is None:
+            raise BadResponse("No file_id in response.")
+
+        # initialize
         self.client = client
-        self.file_id = response.file_id
-        self._pipeline_response = self._initial_response = initial_response
+        self.file_id = response["file_id"]
+        self._initial_response = initial_response
         self._deserialization_callback = deserialization_callback
+        self._resource = None
+        self._finished = False
+        
+        # create the command which will be polled against as the status monitor
         self._command = functools.partial(self.client.get_upload_file, file_id=self.file_id, **self._kwargs)
+        
+        # set initial status
+        self._status = "InProgress"
 
+        
     def status(self) -> str:
-        """Return the current status as a string.
+        """Should return the current status as a string.
+        This is what is returned to the user when status() is called on the LROPoller.
```

Comment:
Do we need to add a note as to whether/how this intertwines with the "polling strategy" for determining status section above?

Reply:
Your comment made me realize that it's probably not totally clear how the strategy plays in with the polling method... We kind of go from 0 to 100 fast. 😆 I added a simple strategy into the custom polling method example so that it hopefully ties in together better.